projects/step-core/src/lib/modules/editable-labels/components/editable-component/editable.component.ts

```diff
@@ -33,19 +36,24 @@ type LabelTemplateAccessor = Mutable<Pick<EditableComponent<any>, 'labelTemplate
   // The provided template in our case is irrelevant
   template: '',
 })
-export abstract class EditableComponent<T> implements ControlValueAccessor, AfterContentInit, OnChanges {
+export abstract class EditableComponent<T> implements ControlValueAccessor {
   protected _elementRef = inject<ElementRef<HTMLElement>>(ElementRef);
   protected _changeDetectorRef = inject(ChangeDetectorRef);
   protected _document = inject(DOCUMENT);
 
-  @Input('labelTemplate') labelTemplateInput!: TemplateRef<{}>;
+  /** @Input('labelTemplate') **/
+  labelTemplateInput = input<TemplateRef<{}> | undefined>(undefined, {
+    alias: 'labelTemplate',
+  });
+
```

Comment:
On first glance this code looks way more complicated than the old syntax. What is the advantage of this?

Reply:
@neogucky In current case converting `labelTemplateInput` input and `labelTemplateDirective` content child to signals, easily allows to compute `labelTemplate` signal and remove `ngAfterContentInit` and `ngOnChanges` hooks.